examples/dataflow-flex-python/README.md

```diff
@@ -0,0 +1,202 @@
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/storage.objectCreator
+
+# Assign Service Account User permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/iam.serviceAccountUser
+
+# Assign BigQuery job ser permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/bigquery.jobUser
+
+# Assign bigquery data editor permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/bigquery.dataEditor
+
+# Assign Artifactory Reader Permissions
+gcloud projects add-iam-policy-binding $PROJECT_ID --member "serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com" --role roles/artifactregistry.reader
+
+# Assign network user permissions on Host project
+gcloud projects add-iam-policy-binding $HOST_PROJECT_ID --member "serviceAccount:service-$PROJECT_NUMBER@dataflow-service-producer-prod.iam.gserviceaccount.com" --role roles/compute.networkUser
+```
+
+#### Setup Cloud Storage
+```
+# Create Cloud Storage bucket for input data
+gcloud storage buckets create gs://$INPUT_BUCKET_NAME --location $LOCATION --project $PROJECT_ID
+
+# Create a bucket for dataflow staging and temp locations
+gcloud storage buckets create gs://$STAGING_BUCKET_NAME --location $LOCATION --project $PROJECT_ID
+
+gsutil iam ch serviceAccount:dataflow-worker-sa@$PROJECT_ID.iam.gserviceaccount.com:roles/storage.legacyBucketWriter gs://$STAGING_BUCKET_NAME
+```
+
+#### Create BQ Dataset
+```
+bq --location=us-central1 mk --dataset $PROJECT_ID:$BQ_DATASET
+```
+
+#### Create ARtifactory registry
+```
+gcloud  artifacts  repositories create $REPO --location $LOCATION --repository-format docker --project $PROJECT_ID
+```
+
+### Build and Push Docker Images for template
+```
+# Build Base Image, all packages will be used from this image when dataflow job runs
+docker build -t $LOCATION-docker.pkg.dev/$PROJECT_ID/$REPO/dataflow-2.40-base:dev .
+
+# Build Image used by launcher to launch the Dataflow job
+cd df-sample-xml-to-bq
+docker build -t $LOCATION-docker.pkg.dev/$PROJECT_ID/$REPO/df-xml-to-bq:dev .
+cd ..
+
+# Push both the images to repo
+docker push $LOCATION-docker.pkg.dev/$PROJECT_ID/$REPO/dataflow-2.40-base:dev
+docker push $LOCATION-docker.pkg.dev/$PROJECT_ID/$REPO/df-xml-to-bq:dev
+```
+
+#### Build Dataflow flex template
+```
+gcloud dataflow flex-template build gs://$INPUT_BUCKET_NAME/dataflow-templates/xml-to-bq.json \
+--image "$LOCATION-docker.pkg.dev/$PROJECT_ID/$REPO/df-xml-to-bq:dev" \
+--sdk-language "PYTHON" \
+--metadata-file df-sample-xml-to-bq/metadata.json \
+--project $PROJECT_ID
+```
+
+#### Upload Sample Data
```

Comment:
And from this section you have a "demo" section to use what you created.

Reply:
done